docs/migration-guide/to-16.md

```diff
@@ -45,6 +45,15 @@ async function lint() {
 
 If you use `stylelint.lint()` to lint files, the `code` property will always be `undefined`.
 
+## Changed Node.js API returned `formatters` object
+
+We've changed the `stylelint.formatters` object in the Node.js API so that every formatter is a `Promise` function.
+
+```diff js
+-stylelint.formatters.json(results);
++stylelint.formatters.json.then(formatter => formatter(results));
```

Comment:
Is it true that the change would *have* to be reverted? This PR is tagged as being `major`/a breaking change (and is part of `v16`); it seems like the intention is that people who rely on features of the existing API would just have to migrate (similar to other breaking changes). Or is there something else I'm missing?

(I could also see us maybe deprecating this feature and removing in a future release, but I'm not sure personally how that would work)

Reply:
Is the logic to pack the most breaking change that we can in v16?
I would think we should instead try to have the least and the most pressing ones only.
Is this change that necessary?

Reply:
> Is this change that necessary?

Yes, as I understand. Unless this change, `formatters/*Formatter.mjs` will never been used. In the code below, `.cjs` are used instead of `.mjs`: ⬇️ 

https://github.com/stylelint/stylelint/blob/3f3206652a60e97631d0677facad69a3bdd197d0/lib/formatters/index.mjs#L8-L16

Changing `require` to `import` means changing to a Promise-based API. Or, do you have any ideas about this difficulty?

Reply:
I checked the code of `stylelint-webpack-plugin`.

https://github.com/webpack-contrib/stylelint-webpack-plugin/blob/64b977d03a9b906acb3b3183b64aaf5458535a98/src/linter.js#L254

Surely, Stylelint 16.0.0 will break the code (loading a formatter), but it can be easily addressed like this:

```diff
-return stylelint.formatters[formatter];
+return Promise.resolve(stylelint.formatters[formatter]);
```

I believe this level of change should be acceptable in the major version.

Reply:
If you are confident that it is not that disruptive, Ill approve.
Too bad async getters don't exist, it would have been cleaner.

Reply:
> Maybe you could use `defineProperty` instead?

Here's the compatible code by using `defineProperty` (`configurable`/`enumerable` are `false` by default):

```js
Object.defineProperty(formatters, 'compact', {
	async get() {
		return import('./compactFormatter.mjs').then((m) => m.default);
	},
	configurable: true,
	enumerable: true,
});
```

I think there are no advantages here since code size increases.